File: Data Structures & Algorithms/invert-a-binary-tree/submission-4.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque
class Solution:
    def invertTree(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        if not root:
            return root

        queue=deque()
        queue.append(root)
        while queue:
            tempRoot=queue.popleft()
            l=tempRoot.left
            tempRoot.left=tempRoot.right
            tempRoot.right=l
            if tempRoot.left:
                queue.append(tempRoot.left)
            if tempRoot.right:
                queue.append(tempRoot.right)
        # bfs TC O(N) AND SC O(N)
        return root

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-4.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func countGoodNode(root *TreeNode, v int) int {
    if root == nil{
        return 0
    }
    var res int
    if root.Val>=v{
        res=1
    }
    v=max(root.Val,v)
    res += countGoodNode(root.Left,v)
    res += countGoodNode(root.Right,v)
    return res
}

func goodNodes(root *TreeNode) int {
    if root == nil{
        return 0
    }

    return countGoodNode(root,root.Val)
}
